packages/gatsby-plugin-google-analytics/src/index.js

```diff
@@ -37,9 +38,10 @@ function OutboundLink(props) {
         }
         if (window.ga) {
           window.ga(`send`, `event`, {
-            eventCategory: `Outbound Link`,
-            eventAction: `click`,
-            eventLabel: props.href,
+            eventCategory: eventCategory || `Outbound Link`,
+            eventAction: eventAction || `click`,
+            eventLabel: eventLabel || props.href,
+            eventValue,
```

Comment:
If we spread `eventValue` it won't be added to the object if it's `undefined`.
```suggestion
            ...eventValue,
```

Reply:
But `eventValue` is a number. Why would you spread it?

Reply:
`window.ga()` accepts a so-called [`fieldsObject`](https://developers.google.com/analytics/devguides/collection/analyticsjs/sending-hits#using_the_ga_command_queue) as property. So we're actually sending an object with key/value pairs to Google Analytics and one of the properties in this object is `eventValue`, which can contain numeric values only (thanks to `PropTypes`).

The spread operator, which is actually synthetic sugar for `Object.assign()`, generates an empty object for `undefined` or `null` values [(see ECMAScript docs)](https://www.ecma-international.org/ecma-262/6.0/index.html#sec-object.assign) and that's why I'm suggesting to use the spread operator.

Reply:
Sorry if my comment was confusing. I know that spreading `undefined` and `null` generates an empty object. But my point is that spreading a number (and `eventValue` is a number) will lead us to the same result:

```js
const eventValue = 5
const fieldsObject = {
  ...eventValue
}
console.log(fieldsObject)
// outputs: {}
```

Reply:
Sorry for the confusion! You're completely right,  spreading numbers will result in an empty object.
My suggestion can be ignored :)